cloud/scope/machine.go

```diff
@@ -338,6 +338,9 @@ func (m *MachineScope) InstanceSpec(log logr.Logger) *compute.Instance {
 				m.ClusterGetter.Name(),
 			),
 		},
+		Params: &compute.InstanceParams{
+			ResourceManagerTags: infrav1.AddResourceManagerTags(context.TODO(), m.GCPMachine.Spec.ResourceManagerTags),
```

Comment:
I think, we should have the merge functionality here for the tags defined in MachineTemplate.Spec and Machine.Spec, with those in Machine.Spec taking precedence.